server/src/repositories/asset.repository.ts

```diff
@@ -800,21 +800,23 @@ export class AssetRepository implements IAssetRepository {
     ],
   })
   getAllForUserFullSync(options: AssetFullSyncOptions): Promise<AssetEntity[]> {
-    const { ownerId, lastCreationDate, lastId, updatedUntil, limit } = options;
-    let builder = this.repository
-      .createQueryBuilder('asset')
-      .leftJoinAndSelect('asset.exifInfo', 'exifInfo')
-      .leftJoinAndSelect('asset.stack', 'stack')
-      .where('asset.ownerId = :ownerId', { ownerId });
+    const { ownerId, isArchived, withStacked, lastCreationDate, lastId, updatedUntil, limit } = options;
+    const builder = this.getBuilder({
+      userIds: [ownerId],
+      exifInfo: true,
+      withStacked,
+      isArchived,
+    });
+
     if (lastCreationDate !== undefined && lastId !== undefined) {
-      builder = builder.andWhere('(asset.fileCreatedAt, asset.id) < (:lastCreationDate, :lastId)', {
+      builder.andWhere('(asset.fileCreatedAt, asset.id) < (:lastCreationDate, :lastId)', {
```

Comment:
Are you sure this works? We assign the (presumably new) builder in other places too

Reply:
Yes, it doesn't need to be re-assigned.

Reply:
Ah ok nice, maybe we could refactor that then in all places?

Reply:
Go for it!

Reply:
#9119 :)